tests/datatype/tests_Int.cpp

```diff
@@ -0,0 +1,57 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+
+#include <doctest/doctest.h>
+#include <rdf4cpp/rdf.hpp>
+
+using namespace rdf4cpp::rdf;
+
+TEST_CASE("Datatype Int") {
+
+    constexpr auto correct_type_iri_cstr = "http://www.w3.org/2001/XMLSchema#int";
+
+    CHECK(std::string(datatypes::xsd::Int::identifier) == correct_type_iri_cstr);
+
+    auto type_iri = IRI(datatypes::xsd::Int::identifier);
+    CHECK(type_iri.identifier() == correct_type_iri_cstr);
+
+    using type = datatypes::xsd::Int::cpp_type;
+
+    CHECK(std::is_same_v<type, int32_t>);
+
+    auto value = 1;
```

Comment:
In some tests you define value using the keyword `auto`, whereas in other test cases you use the `cpp_type` of the datatype. Is there a reason for this?